refactor(behavior_path_avoidance_by_lane_change_module): prefix package and namespace with autoware_

esteve opened the pull request:
## Description

<!-- Write a brief description of this PR. -->

This PR adds the `autoware_` prefix to the package and puts headers in the `autoware` namespace.

Part of:
- https://github.com/autowarefoundation/autoware/issues/4569

## Tests performed

<!-- Describe how you have tested this PR. -->
<!-- Although the default value is set to "Not Applicable.", please update this section if the type is either [feat, fix, perf], or if requested by the reviewers. -->

Not applicable.

## Effects on system behavior

<!-- Describe how this PR affects the system behavior. -->

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

esteve commented:
@xmfcx @mitsudome-r @rej55 @satoshi-ota @shmpwk @TomohitoAndo @tkimura4 @zulfaqar-azmi-t4

This is now ready for review, please have a look at it when you have a moment. Thanks.

esteve commented:
I've reverted any performance or any other improvements detected by clang-tidy, the CI will now fail because of the clang-tidy check, but now the PR only conntains the prefix changes.

esteve commented:
Back to draft because I'm going to rename folders and move headers to a separate folder (see https://github.com/autowarefoundation/autoware/issues/4569#issuecomment-2097690750)

esteve commented:
@xmfcx this PR is ready for review again.

xmfcx commented on planning/autoware_behavior_path_avoidance_by_lane_change_module/src/scene.cpp:
```diff
@@ -32,9 +32,12 @@
 #include <optional>
```
```suggestion
#include <algorithm>
#include <optional>
```

To pass https://results.pre-commit.ci/run/github/429651616/1715682762.0qguD79LT5yQcQVUZwVcpQ

satoshi-ota commented:
Hi @esteve . I missed your PR and created https://github.com/autowarefoundation/autoware.universe/pull/7309 accidentally. Could you fix build fail? After that, I'll approve your PR.

esteve commented:
@satoshi-ota done, I've fixed all the build issues. I've also moved the header to src so they are not exported (there are no downstream packages that depend on this package), and put all the code in the `autoware::` namespace

satoshi-ota reviewed (approved):
LGTM!
